python/python/bystro/prs/tests/test_prs_dosage.py

```diff
@@ -0,0 +1,277 @@
+
+
+def test_prs_one_batch(create_feather_file):
+    """
+    Test PRS calculations with Pandas
+    This will be less memory efficient than operating on the Feather file directly
+    """
+    file_path = create_feather_file
+    prs_scores = calculate_prs_in_batches(file_path, weights)
+    prs_scores_pandas = calculate_prs_in_pandas(file_path, weights)
+
+    # deep equality check that these two dictionaries contain the same keys and values
+    assert prs_scores == prs_scores_pandas
+
+    assert len(prs_scores) == 3
+    for sample in prs_scores:
+        assert prs_scores[sample] > 0
+
+    print(prs_scores)
+
+
+def test_prs_multiple_batches(create_feather_file_with_multiple_batches):
+    """
+    Test PRS calculations with Pandas
+    This will be less memory efficient than operating on the Feather file directly
+    """
+    file_path, weights = create_feather_file_with_multiple_batches
+    prs_scores = calculate_prs_in_batches(file_path, weights)
+
+    assert len(prs_scores) == 3
+    for sample in prs_scores:
+        assert prs_scores[sample] > 0
+
+    print(prs_scores)
+
+
+def test_prs_columnar_one_batch(create_feather_file_with_multiple_batches):
+    file_path, weights = create_feather_file_with_multiple_batches
+
+    expected = calculate_prs_in_pandas(file_path, weights)
+
+    mask = pc.field("locus").isin(weights.index)
+
+    my_dataset = ds.dataset([file_path], format="arrow")
+    samples = set(my_dataset.schema.names) - set(["locus"])
+
+    filtered_data = my_dataset.filter(mask)
+
+    prs_scores = {}
+    for sample in samples:
+        sample_genotypes = filtered_data.to_table(["locus", sample]).to_pandas()
+        sample_genotypes = sample_genotypes.set_index("locus")
+
+        weights_subset = weights.loc[sample_genotypes.index]
+        prs_scores_batch = sample_genotypes.multiply(weights_subset, axis="index").sum()
+        prs_scores[sample] = prs_scores_batch.item()
+
+    assert prs_scores == expected
+    print(prs_scores)
+
+
+def test_prs_columnar_batches(create_feather_file_with_multiple_batches):
+    file_path, weights = create_feather_file_with_multiple_batches
+    mask = pc.field("locus").isin(weights.index)
+
+    my_dataset = ds.dataset([file_path], format="arrow")
+    samples = set(my_dataset.schema.names) - set(["locus"])
+
+    filtered_data = my_dataset.filter(mask)
+
+    prs_scores: dict[str, float] = {}
+    for batch in filtered_data.to_batches():
+        for sample in samples:
+            sample_genotypes = batch.select(["locus", sample]).to_pandas()
+            sample_genotypes = sample_genotypes.set_index("locus")
+
+            weights_subset = weights.loc[sample_genotypes.index]
+            prs_scores_batch = sample_genotypes.multiply(weights_subset, axis="index").sum()
+
+            if sample in prs_scores:
+                prs_scores[sample] += prs_scores_batch
+            else:
+                prs_scores[sample] = prs_scores_batch
+
+    print(prs_scores)
```

Comment:
Will all of these tests that are performance tests and examples generating dummy prs_scores be running every time? Shouldn't we just do these kinds of tests one time and then decide the best method? Just want some clarification on the purpose of the tests here/whether they're temporary while we're deciding on the design details of this module since they might kind of visually clog up the test suite outputs versus just having unit tests or integration tests in the codebase.

Reply:
These tests cumulatively take 

```py
================================================================== 7 passed in 0.73s ==================================================================

real    0m1.140s
user    0m1.172s
sys     0m0.280s
```

to run

Which ones do you want me to remove?

Also pytest never outputs the print messages. You have to explicitly ask for them.

Reply:
I've cleaned up the tests, removing print statements, and doing a better job demonstrating the equivalence of different approaches.

Reply:
I'll also mention that a large benefit of a testing suite is as documentation. Here we are demonstrating different ways that someone can use our output format to generate PRS scores

Reply:
Ok gotcha, didn't realize those were optional outputs. Mostly just wanted a clarification on general purpose of tests wrt which tests we want to have in the codebase vs running in advance of PRing to figure out a design choice outside of the codebase.

Reply:
Added the benchmark and clarified the tests